### PR TITLE
Added Elo

### PR DESCRIPTION
**Pull request by yonbergman**
- fixed end match (the issue was `Date.now` vs `new Date`

### CODE REVIEW

File: cloud/main.js [hunk 1]
```diff
@@ -1,7 +1,60 @@
...
+    return Elo.expected(match.get('playerA').get('person').get('rank'),
+                        match.get('playerB').get('person').get('rank'));
+  },
+  adjustMatchResults: function(match){
+    if (!match) return null;
+    var personA = match.get('playerA').get('person');
+    var rankA = personA.get('rank');
+    var personB = match.get('playerB').get('person');
+    var rankB = personB.get('rank');
+    var playerAdidWin = match.get('winner') == 'playerA';
```
Comment (udiWertheimer) [inline cloud/main.js]: I prefer `playerADidWin` over `playerAdidWin` (was trying to figure out what "adid" was). Just nitpicking, you decide

**Comment by udiWertheimer** [comment]
:+1:

File: cloud/main.js [hunk 2]
```diff
@@ -1,7 +1,60 @@
 _ = require('underscore');
-var Character = Parse.Object.extend("Character", {
 
+var Elo = {
+  startRank: 1200,
+  K: 20,
```
Comment (giladgo) [inline cloud/main.js:5]: `K`?

File: cloud/main.js [hunk 3]
```diff
@@ -1,7 +1,60 @@
...
+  expected: function(rankA, rankB){
+    return 1/(1 + Math.pow(10, (rankB-rankA)/400));
```
Comment (giladgo) [inline cloud/main.js:7]: Is `400` a magic number?

File: cloud/main.js [hunk 4]
```diff
@@ -363,3 +471,21 @@ Parse.Cloud.define("endMatch", function(request, response) {
...
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+Parse.Cloud.beforeSave("Person", function(request, response) {
+  var person = request.object;
+  if (person.existed()) {
+    response.success();
+  } else {
+    person.set('rank', Elo.startRank);
+    response.success();
```
Comment (giladgo) [inline cloud/main.js:489]: :+1:

**Comment by giladgo** [comment]
:light_rail:

**Pull request closed without merging**